src/script/script.js

```diff
@@ -0,0 +1,402 @@
+      <h1 class="header">RS-School task Virtual Keyboard</h1>
+      <p class="description">Клавиатура создана в операционной системе Windows</p>
+      <p class="description">Переключения языка клавиатурой: левыe ctrl + alt</p>
+      <p class="description">Переключение языка мышкой: кнопка с иконкой глобуса</p>`;
+    this.rootElement.append(this.header);
+  }
+
+  generateTextArea() {
+    if (!this.rootElement) this.setRootElement();
+    let textArea = document.createElement('div');
+    textArea = document.createElement('div');
+    textArea.classList.add('textarea-wrapper');
+    textArea.innerHTML = `
+      <textarea class="textarea" id="textarea" rows="5" cols="50" autofocus></textarea>`;
+    this.rootElement.append(textArea);
+    this.textArea = document.querySelector('#textarea');
+    this.textArea.value = '';
+  }
+
+  generateKeyboard() {
+    if (!this.rootElement) this.setRootElement();
+
+    const rows = this.data || this.defaultSet;
+
+    this.keyBoard = document.createElement('div');
+    this.keyBoard.classList.add('keyboard');
+    rows.forEach(elem => {
+      this.keyBoard.append(this.generateKeyboardRow(elem));
+    });
+    this.rootElement.append(this.keyBoard);
+    this.keyBoard = document.querySelector('.keyboard');
+    this.keys = document.querySelectorAll('.symbol');
+    this.languageButton = document.querySelector('.ContextMenu span');
+
+    this.pressed = new Set();
+
+    this.textArea.addEventListener("click", () => {
+      this.currentPosition = this.getCaretPosition();
+    })
+
+    document.addEventListener("keydown", (event) => {
+      if (event.code === "F5") return;
+      event.preventDefault();
+      this.handleButtonDown(event.code)
+    });
+    document.addEventListener("keyup", (event) => {
+      this.handleButtonUp(event.code);
+    });
+
+    this.keyBoard.addEventListener("mousedown", (event) => {
+      this.handleKeyboardMousedown(event);
+    });
+    this.keyBoard.addEventListener("mouseup", () => {
+      this.handleKeyboardMouseup(this.mouseClicked);
+    });
+  }
+  ///////////////////////////////////////////////////////////////
+  generateKeyboardRow(array) {
+    if (localStorage.keyboardLanguage !== "ru" && localStorage.keyboardLanguage !== "en") {
+      localStorage.keyboardLanguage = "ru";
+    }
+    let row = document.createElement('div');
+    row.classList.add('keyboard--row');
+    array.forEach(elem => {
+      row.append(this.generateKeyboardButton(elem));
+    });
+    return row;
+  }
+
+  generateKeyboardButton(obj) {
+    let btn = document.createElement('div');
+    btn.classList.add("keyboard--btn", obj.className);
+    if (obj.symbol) {
+      btn.classList.add("symbol");
+      let current = localStorage.keyboardLanguage === "ru" ? obj.ru.down : obj.en.down;
+      btn.dataset.value = current;
+    };
+
+    let output = !obj.symbol ? `<span>${obj.value}</span>` :
+      `<span class="ru ${localStorage.keyboardLanguage === "ru" ? "" : "hidden"}">
```

Comment:
Yes, I know about this opportunity. In this case, I initially adopted an unsuccessful application architecture, which implies changing the transparency property of the button’s children, instead of placing all the data in the data-set attributes and retrieving them from there to replace the contents of the button. I tried to fix it, however, for this I need to completely rewrite ALL the application from scratch, including completely reworking the input data, because It is on such a DOM structure that the entire application logic is tied - from drawing buttons to processing all events on these buttons. Any changes to this structure will break the application. At the moment, I understand that it would be much better to make a set of ordinary <button> elements without a nested structure. But in this case, I have to make the application in full from scratch.

Reply:
All completely and comprehensive refactored